### PR TITLE
Add gazebo dual rotor tailsitter.

### PR DESCRIPTION
**Pull request by grahamjamesaddis**
A twin rotor gazebo tailsitter model as per vtol_duo_tailsitter:

https://docs.px4.io/master/en/airframes/airframe_reference.html#vtol-duo-tailsitter

Once this is merged I'll update the PX4-Autopilot when this merge is complete.
Preliminary branch available here:
https://github.com/grahamjamesaddis/PX4-Autopilot/tree/add_tailsitter_duo

### CODE REVIEW

**Review by Jaeyoung-Lim — COMMENTED** [review]
Thanks for the contribution!

Could you confirm that this model already flies in SITL? If so, it would be great it would be possible to test with the firmware changes.

File: models/tailsitter_duo/model.config [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
+<model>
+  <name>Duo Tailsitter</name>
+  <version>1.0</version>
+  <sdf version='1.6'>tailsitter_duo.sdf</sdf>
+
+  <author>
+   <name>Roman Bapst</name>
+   <email>roman@px4.io</email>
```
Comment (Jaeyoung-Lim) [inline models/tailsitter_duo/model.config:9]: I don't think this model was written by Roman Bapst. Or is it?

Comment (grahamjamesaddis) [reply models/tailsitter_duo/model.config:9]: I can't really claim credit, I removed two motors, moved and renamed the others, the real work was already done.

It flies, but is not stable, which is why I refer to the changes it to support in PX4 as preliminary.

I'm happy to try it against another branch.

Comment (Jaeyoung-Lim) [reply models/tailsitter_duo/model.config:9]: @grahamjamesaddis How can I test the vehicle? Could you provide instructions in this PR?

**Comment by Jaeyoung-Lim** [comment]
One big aspect of this model that is missing (which is why we didn't have the duo tailsitter for a long time) is that the rotor slip stream effect is not accounted for. Therefore I am not sure how this model would be able to fly with the slip stream modelled.

File: models/tailsitter_duo/model.config (hunk 1 above)
Comment (grahamjamesaddis) [reply models/tailsitter_duo/model.config:9]: At high level:

Pull https://github.com/grahamjamesaddis/PX4-Autopilot/tree/add_tailsitter_duo

Use https://github.com/grahamjamesaddis/PX4-SITL_gazebo/tree/add_tailsitter_duo for sitl_gazebo

Build with command:
make px4_sitl gazebo_tailsitter_duo

**Comment by grahamjamesaddis** [comment]
> One big aspect of this model that is missing (which is why we didn't have the duo tailsitter for a long time) is that the rotor slip stream effect is not accounted for. Therefore I am not sure how this model would be able to fly with the slip stream modelled.

That looks like a good reason for the crash:

Here's a log from today.

https://logs.px4.io/plot_app?log=403bcbe9-2b44-4907-a42e-50c4994d8cd8

**Comment by Jaeyoung-Lim** [comment]
@grahamjamesaddis Thanks for the logs and branches. I am not sure if this is worth merging since it is not flying properly. Would you be able to fix the issues with the tailsitter propwash modeling?

**Comment by grahamjamesaddis** [comment]
Given that I don't have a real model to validate any changes against, I'll put this on hold for the moment.

It's quite straightforward to re-introduce if needed.

**Comment by xdwgood** [comment]
We have real dual rotor drones flying, can we push this PR together?

**Comment by tstastny** [comment]
> We have real dual rotor drones flying, can we push this PR together?

@xdwgood , as @Jaeyoung-Lim mentioned - you'll need to model the slip stream over the wings, otherwise the flaps will not generate any force in hover, and the vehicle behavior in cruise will also not consider the slip stream.

This is something we've discussed a few times - but never arrived to a clear path forward (how to detect if there are lifting surfaces behind a propeller in gazebo, and how to then model the slip stream velocity getting superposed). If you want to implement a plugin for this, please do share your thoughts on how you may want to design it!

**Comment by mengchaoheng** [comment]
@tstastny [PR](https://github.com/PX4/PX4-SITL_gazebo-classic/pull/1083) is in progress.

**Pull request closed without merging**